dll/cpl/hotplug/hotplug.c

```diff
@@ -344,6 +344,78 @@ ShowContextMenu(
 }
 
 
+static
+DWORD
+GetSelectedDeviceInst(
+    _In_ HWND hwndDeviceTree)
+{
+    HTREEITEM hTreeItem;
+    TVITEMW item;
+
+    hTreeItem = TreeView_GetSelection(hwndDeviceTree);
+    if (hTreeItem == NULL)
+        return 0;
+
+    ZeroMemory(&item, sizeof(item));
+    item.mask = TVIF_PARAM;
+    item.hItem = hTreeItem;
+
+    TreeView_GetItem(hwndDeviceTree, &item);
+
+    return item.lParam;
+}
+
+static
+int
+_DevicePropertiesW(
+    _In_ HWND hWndParent,
+    _In_ LPCWSTR lpMachineName,
+    _In_ LPCWSTR lpDeviceID,
+    _In_ BOOL bShowDevMgr)
+{
+    typedef int (WINAPI *PFDEVICEPROPERTIESW)(HWND, LPCWSTR, LPCWSTR, BOOL);
+    HMODULE hDll;
+    PFDEVICEPROPERTIESW pFunc;
+    int ret = -1;
+
+    hDll = LoadLibraryW(L"devmgr.dll");
+    if (hDll == NULL)
+        return -1;
+
+    pFunc = (PFDEVICEPROPERTIESW)GetProcAddress(hDll, "DevicePropertiesW");
+    if (pFunc != NULL)
+        ret = pFunc(hWndParent, lpMachineName, lpDeviceID, bShowDevMgr);
+
+    FreeLibrary(hDll);
+
+    return ret;
+}
+
+static
+VOID
+ShowDeviceProperties(
+    _In_ HWND hwndParent,
+    _In_ DWORD DevInst)
```

Comment:
This should be a `DEVINST` parameter. (If using that type makes compilaton fail, then you may need to include `cfgmgr32.h` .)

EDIT: I see that the existing code (above in that file) uses DWORD, but let's use the actual correct types when possible.

Reply:
Done.